test_conformance/non_uniform_work_group/TestNonUniformWorkGroup.cpp

```diff
@@ -733,7 +756,15 @@ int TestNonUniformWorkGroup::runKernel () {
   err = clSetKernelArg(_testKernel, 1, localArraySize, NULL);
   test_error(err, "clSetKernelArg failed");
 
-  size_t globalBufferSize = adjustGlobalBufferSize(_numOfGlobalWorkItems*sizeof(cl_uint));
+  size_t globalBufferSize =
+      adjustGlobalBufferSize(_numOfGlobalWorkItems * sizeof(cl_uint));
+  if (globalBufferSize != _numOfGlobalWorkItems * sizeof(cl_uint))
+  {
+      // Adjust _globalSize and _numOfGlobalWorkItems accordingly if global
+      // buffer size is adjusted.
+      size_t maxNumOfGlobalWorkItems = globalBufferSize / sizeof(cl_uint);
+      adjustGlobalSize(maxNumOfGlobalWorkItems);
```

Comment:
This function can modify _numOfGlobalWorkItems but globalBufferSize will stay the same. In the result we have again
`_numOfGlobalWorkItems *  sizeof(cl_uint) != globalBufferSize `
Am I right ?

Reply:
Yes, you're right. It won't cause new issue in testBarriers because kernel only accesses the global buffer in range [0, _numOfGlobalWorkItems)

Reply:
Done. Update globalBufferSize again to _numOfGlobalWorkItems * sizeof(cl_uint)